packages/cli/src/commands/init/init.ts

```diff
@@ -56,18 +57,64 @@ interface TemplateOptions {
   skipInstall?: boolean;
   packageName?: string;
   installCocoaPods?: string | boolean;
+  replaceDirectory?: string | boolean;
 }
 
 function doesDirectoryExist(dir: string) {
   return fs.existsSync(dir);
 }
 
-async function setProjectDirectory(directory: string) {
-  if (doesDirectoryExist(directory)) {
+function getConflictsForDirectory(directory: string) {
+  return readdirSync(directory);
+}
+
+async function setProjectDirectory(
+  directory: string,
+  replaceDirectory: string,
+) {
+  const directoryExists = doesDirectoryExist(directory);
+
+  if (replaceDirectory === 'false' && directoryExists) {
     throw new DirectoryAlreadyExistsError(directory);
   }
 
+  let deleteDirectory = false;
+
+  if (replaceDirectory === 'true' && directoryExists) {
+    deleteDirectory = true;
+  } else if (directoryExists) {
+    const conflicts = getConflictsForDirectory(directory);
+
+    if (conflicts.length > 0) {
+      logger.warn(
+        `The directory ${chalk.bold(
+          directory,
+        )} contains files that will be overwritten:`,
+      );
+
+      for (const conflict of conflicts) {
+        logger.log(`  ${conflict}`);
+      }
```

Comment:
this should be the part of the `logger.warn` so that the connected messages are kept together. In case someone wanted to only inspect "warn"s they wouldn't see relevant data for this message